### ISSUE TITLE
[3.3.5/Hotswap] Crash in XT002

### ISSUE THREAD

**Issue by Keader** [TrinityCore/TrinityCore#18565]
**This Crash only happens with Dynamic Build.**

**Description:** Using Hotswap system in XT002, after reload scripts server crash.

**Steps to reproduce the problem:**

1. Go to xt002 (.go crea id 33293)
2. Change something in xt002 script (with hotswap)
3.  Wait hotswap reload scripts

**Branch(es):** 3.3.5
**TC rev. hash/commit:**  554c28d56b9fdbd70f9c1376d0cad5e72a357c87
**OS:** Win10

This crash looks similar to #17557
**When crash happens, crashs folder remains empty (no log is generated)**
When crash happens, show this message in console:

> C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xtree(238) : Assertion failed: map/set iterator not dereferencable
> C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xtree(239) : Assertion failed: "Standard C++ Libraries Out of Range" && 0

Callstack: https://gist.github.com/Keader/b458e67b2aa2511bfe77b4c96bb84d1a

**Comment by sirikfoll** [comment]
I'm having the same crash, creature responsable is [Pyrite Safety Container](http://wotlk.openwow.com/npc=33218)

**Comment by Naios** [comment]
Yeah, in some cases when accessing unloaded memory from the DLL a crashlog isn't generated,
however I can attach MSVC directly.
@sirikfoll please open another ticket for this issue regardless if you can provide a crashlog or not.

I'll take a look into those issues as soon as I'm done with my bachelor's thesis.

**Comment by Naios** [comment]
I took a look on this issue and I can confirm the crash.
However, I'm not able to figure it out because really strange things are going on there as shown in the screenshot below:

![screenshot_11111](https://cloud.githubusercontent.com/assets/1146834/24048671/e3040f96-0b29-11e7-8ae7-508e5d4cd074.png)

Maybe anyone has an idea about this.
